roofit/roostats/src/MarkovChain.cxx

```diff
@@ -223,14 +223,14 @@ THnSparse* MarkovChain::GetAsSparseHist(RooAbsCollection* whichVars) const
    std::vector<Double_t> max(dim);
    std::vector<Int_t> bins(dim);
    std::vector<const char *> names(dim);
-   TIterator* it = axes.createIterator();
-   for (Int_t i = 0; i < dim; i++) {
-      RooRealVar * var = dynamic_cast<RooRealVar*>(it->Next() );
+   Int_t i = 0;
+   for (auto const *var : static_range_cast<RooRealVar *>(axes)) {
```

Comment:
Please remove the line with the assert down below now.

Here is something you should know about `static_range_cast`: in debug mode, it does an `assert(dynamic_cast<...)` internally. So you don't need to do that yourself.